src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -178,6 +178,10 @@ protected void processFiltered(File file, List<String> lines) throws CheckstyleE
                     final FileContents contents = new FileContents(text);
                     final DetailAST rootAST = parse(contents);
 
+                    // Mutation which removes this line of code survives.
+                    // And it looks like its correct, cause the only non-test place which call
+                    // this method is AbstractFileSetCheck.process(). And "process" method clears
+                    // message collector before call to "processFiltered".
                     getMessageCollector().reset();
```

Comment:
it was added during comments parsing introduction - https://github.com/Nimfadora/checkstyle/blame/2483ddc5a4d286891da1922cadf9e66547a7ec48/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L179
I do not see a reason, it is better to manage messageCollector only in class thats is own it - AbstractFileSetCheck.

@Nimfadora , please remove that dead/extra code.
Please beware that there will be more cases like this, please report them in advance